modules/020-deckhouse/hooks/update_deckhouse_image.go

```diff
@@ -108,6 +108,21 @@ type deckhousePodInfo struct {
 	Ready     bool   `json:"ready"`
 }
 
+func (dpi deckhousePodInfo) isBootstrapImage() bool {
+	arr := strings.Split(dpi.Image, ":")
```

Comment:
A plain split is not ok here because it does not allow registries with a port, e.g., `registry.test.com:5000/deckhouse:alpha` (valid private registry address).

There are two options to parse the tag correctly:
1. `ParseReference` method from the `go-containerregistry` library.
2. A method copied from Docker https://github.com/deckhouse/deckhouse/blob/a7535a49dcee6ac56ff8ba9fa7ea83106e01cabb/global-hooks/discovery/deckhouse_registry.go#L169

And we also need a test for it.

Reply:
Makes sense. Thank you.
But as we need only tag here we can use LastIndex for figuring it out

Reply:
The last index can also be a problem. Example:
```
registry.deckhouse.io:alpha@sha256:abcdefg
```